test/test_confighelper.py

```diff
@@ -156,7 +156,7 @@ def test_hostname_loaded_from_file_is_not_overriden_by_metadata(self):
     def test_instance_id_is_used_as_hostname_if_not_specified_in_config(self):
         expected_host = "Valid_Instance_ID"
         self.server.set_expected_response(expected_host, 200)
-        self.config_helper = ConfigHelper(config_path=ConfigHelperTest.VALID_CONFIG_WITH_CREDS_ONLY, metadata_server=self.server.get_url())
+        self.config_helper = ConfigHelper(config_path=ConfigHelperTest.VALID_CONFIG_WITH_CREDS_AND_REGION, metadata_server=self.server.get_url())
```

Comment:
What's the purpose of this change?

Reply:
To be honest I wasn't sure how this test passed before. Without providing the region, this results in two calls to the metadata service, but we are only mocking the result of one of them.

Reply:
I see, sounds good.

Reply:
Ah, I believe I know why it worked - before, it would just truncate the hostname as if it were the AZ string. With my change, however, it tried to parse the hostname as JSON, which fails. That exception must get suppressed and the region is treated as an empty string.

I think this change is necessary unless we enhance the fake metadata service to be able to mock different responses to different URLs (or a sequence of responses).